fwprovider/cluster/metrics/resource_metrics_server.go

```diff
@@ -0,0 +1,373 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var (
+	_ resource.Resource                = &metricsServerResource{}
+	_ resource.ResourceWithConfigure   = &metricsServerResource{}
+	_ resource.ResourceWithImportState = &metricsServerResource{}
+)
+
+type metricsServerResource struct {
+	client *metrics.Client
+}
+
+// NewMetricsServerResource creates new metrics server resource.
+func NewMetricsServerResource() resource.Resource {
+	return &metricsServerResource{}
+}
+
+func (r *metricsServerResource) Metadata(
+	_ context.Context,
+	req resource.MetadataRequest,
+	resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_metrics_server"
+}
+
+func (r *metricsServerResource) Configure(
+	_ context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	cfg, ok := req.ProviderData.(config.Resource)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *proxmox.Client, got: %T", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = cfg.Client.Cluster().Metrics()
+}
+
+func (r *metricsServerResource) Schema(
+	_ context.Context,
+	_ resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Manages PVE metrics server.",
+		Attributes: map[string]schema.Attribute{
+			"id": attribute.ID(),
+			"name": schema.StringAttribute{
+				Description: "Unique name that will be ID of this metric server in PVE.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"disable": schema.BoolAttribute{
+				Description: "Set this to `true` to disable this metric server.",
+				Optional:    true,
+				Default:     nil,
+			},
+			"mtu": schema.Int64Attribute{
+				Description: "MTU (maximum transmission unit) for metrics transmission over UDP. " +
+					"If not set, PVE default is `1500` (allowed `512` - `65536`).",
+				Validators: []validator.Int64{int64validator.Between(512, 65536)},
+				Optional:   true,
+				Default:    nil,
+			},
+			"port": schema.Int64Attribute{
+				Description: "Server network port.",
+				Required:    true,
+				Validators:  []validator.Int64{int64validator.Between(1, 65536)},
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+			},
+			"server": schema.StringAttribute{
+				Description: "Server dns name or IP address.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
```

Comment:
Does changing server / port actually require the resource replace? Does API support update for the service?

Reply:
I can update them in UI though
<img width="611" alt="Screenshot 2025-01-28 at 7 47 09 PM" src="https://github.com/user-attachments/assets/f3d0c7d1-e8e2-4210-8925-dcc9f270764f" />

Reply:
good point! fixed